README.markdown

```diff
@@ -4932,7 +4932,9 @@ ngx.req.set_body_data
 
 Set the current request's request body using the in-memory data specified by the `data` argument.
 
-If the current request's request body has not been read, then it will be properly discarded. When the current request's request body has been read into memory or buffered into a disk file, then the old request body's memory will be freed or the disk file will be cleaned up immediately, respectively.
+If the request body has not been read yet, call [ngx.req.read_body](#ngxreqread_body) first (or turn on [lua_need_request_body](#lua_need_request_body) to force this module to read the request body. This is not recommended however). Additionally, the request body must not have been previously discarded by [ngx.req.discard_body](#ngxreqdiscard_body).
```

Comment:
Pedantically minor grammar comment: imo the first clause of this sentence reads better as

"If the request body has not yet been read"

Reply:
> Additionally, the request body must not have been previously discarded by [ngx.req.discard_body](#ngxreqdiscard_body).

What happens if `ngx.req.discard_body` has been called? Would be nice to have documentation denoting the specific consequences.

Reply:
1st comment: Indeed, but it stays consistent with the grammar already used in other sections of the documentation mentioning `ngx.req.read_body`, and I did not want to apply drastic changes to it at the risk of seeing this PR rejected. I'll defer to @agentzh.